Sources/SpeziHealthKit/Queries/QueryAnchor.swift

```diff
@@ -0,0 +1,50 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import HealthKit
+
+
+/// Used by anchor queries to keep track of the last-seen state of the HealthKit database.
+public struct QueryAnchor<Sample: _HKSampleWithSampleType>: Hashable, Codable, Sendable {
```

Comment:
@PSchmiedmayer i'd appreciate your feedback on this, and whether you think we should keep this type, or just use a plain `HKQueryAnchor` instead.

my idea here was to make the anchors directly Codable, so that users can store (and re-use) them using a system  of their choosing.
i also figured that making the anchor generic over a sample type would make it more difficult to accidentally use the wrong anchor somewhere (i'm not really sure whether this part was actually a good idea, tbh)

there's three paths here:
1. keep it as-is
2. keep it but make it non-generic
3. just use a direct `HKQueryAnchor` instead

what's your take?

Reply:
I would vote for option 2.